### ISSUE TITLE
go 1.12: panic: runtime error: hash of unhashable type model.IPPoolKey

### ISSUE THREAD

**Issue by dberg1** [projectcalico/felix#1965]
There is a bug in calc/event_sequencer.go:

```
func (buf *EventSequencer) OnIPPoolUpdate(key model.IPPoolKey, pool *model.IPPool) {
	log.WithFields(log.Fields{
		"key":  key,
		"pool": pool,
	}).Debug("IPPool update")
	buf.pendingIPPoolDeletes.Discard(key)       <<< wrong key
	cidr := ip.CIDRFromCalicoNet(key.CIDR)
	buf.pendingIPPoolUpdates[cidr] = pool
}
```
model.IPPoolKey is not hashable.
The code should be:

```
	cidr := ip.CIDRFromCalicoNet(key.CIDR)
	buf.pendingIPPoolDeletes.Discard(cidr)      <<< correct key
	buf.pendingIPPoolUpdates[cidr] = pool
```

The panic only happens when compiled with go 1.12 as earlier versions of go did not detect this type of error when the map is empty:

```
# cat > hash.go << EOF
package main
 
func main() {
        m := make(map[interface{}]bool)
        b := []byte{1, 2, 3, 4}
        delete(m, b)
}
EOF
```
With go 1.12:
```
# go run hash.go
panic: runtime error: hash of unhashable type []uint8

goroutine 1 [running]:
main.main()
        /tmp/hash.go:6 +0xdc
```
With go 1.11:
```
# go run hash.go
```

**Comment by dberg1** [comment]
```
panic: runtime error: hash of unhashable type model.IPPoolKey

goroutine 75 [running]:
github.com/projectcalico/node/vendor/github.com/projectcalico/libcalico-go/lib/set.mapSet.Discard(0xa0001000018c3c0, 0x110275e60, 0xa000100007d4f00)
        /go/src/github.com/projectcalico/node/vendor/github.com/projectcalico/libcalico-go/lib/set/set.go:85 +0x50
github.com/projectcalico/node/vendor/github.com/projectcalico/felix/calc.(*EventSequencer).OnIPPoolUpdate(0xa000100002f0000, 0xa000100006a442c, 0x4, 0x4, 0xa000100006a4428, 0x4, 0x4, 0xa0001000035a6c0)
        /go/src/github.com/projectcalico/node/vendor/github.com/projectcalico/felix/calc/event_sequencer.go:477 +0x290
github.com/projectcalico/node/vendor/github.com/projectcalico/felix/calc.(*DataplanePassthru).OnUpdate(0xa00010000425da0, 0x110401280, 0xa0001000066e900, 0x11009c660, 0xa0001000035a6c0, 0xa000100006a440c, 0x3, 0x0, 0x100000000000000, 0x110179060)
        /go/src/github.com/projectcalico/node/vendor/github.com/projectcalico/felix/calc/dataplane_passthru.go:76 +0x260
github.com/projectcalico/node/vendor/github.com/projectcalico/felix/dispatcher.updateHandlers.DispatchToAll(0xa000100000100f8, 0x1, 0x1, 0x110401280, 0xa0001000066e900, 0x11009c660, 0xa0001000035a6c0, 0xa000100006a440c, 0x3, 0x0, ...)
        /go/src/github.com/projectcalico/node/vendor/github.com/projectcalico/felix/dispatcher/dispatcher.go:45 +0xac
github.com/projectcalico/node/vendor/github.com/projectcalico/felix/dispatcher.(*Dispatcher).OnUpdate(0xa00010000424420, 0x110401280, 0xa0001000066e900, 0x11009c660, 0xa0001000035a6c0, 0xa000100006a440c, 0x3, 0x0, 0x100000000000000, 0xa0001000045fca8)
        /go/src/github.com/projectcalico/node/vendor/github.com/projectcalico/felix/dispatcher/dispatcher.go:102 +0x318
github.com/projectcalico/node/vendor/github.com/projectcalico/felix/dispatcher.(*Dispatcher).OnUpdates(0xa00010000424420, 0xa000100005cb000, 0x38, 0x38)
        /go/src/github.com/projectcalico/node/vendor/github.com/projectcalico/felix/dispatcher/dispatcher.go:80 +0x120
github.com/projectcalico/node/vendor/github.com/projectcalico/felix/calc.(*AsyncCalcGraph).loop(0xa000100005d28a0)
        /go/src/github.com/projectcalico/node/vendor/github.com/projectcalico/felix/calc/async_calc_graph.go:145 +0x2e4
created by github.com/projectcalico/node/vendor/github.com/projectcalico/felix/calc.(*AsyncCalcGraph).Start
        /go/src/github.com/projectcalico/node/vendor/github.com/projectcalico/felix/calc/async_calc_graph.go:232 +0xd4
```

**Comment by fasaxc** [comment]
Fixed by #1981
